packages/material-renderers/src/layouts/ExpandPanelRenderer.tsx

```diff
@@ -296,12 +299,35 @@ export const withContextToExpandPanelProps = (
     props,
   }: JsonFormsStateContext & ExpandPanelProps) {
     const dispatchProps = ctxDispatchToExpandPanelProps(ctx.dispatch);
-    const { childLabelProp, schema, path, index, uischemas } = props;
+    const { childLabelProp, schema, rootSchema, path, index, uischemas } =
+      props;
     const childPath = composePaths(path, `${index}`);
     const childData = Resolve.data(ctx.core.data, childPath);
-    const childLabel = childLabelProp
-      ? get(childData, childLabelProp, '')
-      : get(childData, getFirstPrimitiveProp(schema), '');
+
+    let childLabel;
+    if (childLabelProp) {
```

Comment:
The label determination should be within a `useMemo`

Reply:
Fixed